ProjFS.Mac/PrjFSKextTests/MockVnodeAndMount.cpp

```diff
@@ -33,11 +38,68 @@ vnode::vnode(const shared_ptr<mount>& mount) :
 {
 }
 
+vnode::vnode(const std::shared_ptr<mount>& mount, VnodeCreationProperties properties) :
+    mountPoint(mount),
+    name(nullptr),
+    inode(properties.inode == UINT64_MAX ? mount->nextInode++ : properties.inode),
+    type(properties.type),
+    parent(properties.parent)
+{
+}
+
 vnode::~vnode()
 {
     assert(this->ioCount == 0);
 }
 
+static string ParentPathString(const string& path)
+{
+    assert(path.length() > 0);
+    size_t lastSlashPos = path.find_last_of('/');
+    if (lastSlashPos == 0)
+    {
+        return "/";
+    }
+    else if (lastSlashPos == path.length() - 1) // path ends in "/"
+    {
+        lastSlashPos = path.find_last_of('/', lastSlashPos - 1);
+    }
+    
+    return path.substr(0, lastSlashPos);
+}
+
+shared_ptr<vnode> mount::CreateVnodeTree(string path, vtype vnodeType)
+{
+    assert(path[0] == '/'); // Only absolute paths allowed
+    string parentPath = ParentPathString(path);
+    PathToVnodeMap::const_iterator found = s_vnodesByPath.find(parentPath);
+    shared_ptr<vnode> parentVnode = path == "/" ? nullptr : (found != s_vnodesByPath.end() ? found->second.lock() : nullptr) ?: this->CreateVnodeTree(parentPath, VDIR);
```

Comment:
While the above line is very concise, it's not easy (for me at least) to read.  Thoughts on making it more verbose for clarity?

Reply:
Yeah, that was a rats' nest. I've cleaned it up.